app/services/bgs/dependent_service.rb

```diff
@@ -33,20 +33,39 @@ def get_dependents
     end
 
     def submit_686c_form(claim)
-      bgs_person = service.people.find_person_by_ptcpnt_id(participant_id)
-
-      bgs_person = service.people.find_by_ssn(ssn) if bgs_person.nil? # rubocop:disable Rails/DynamicFindBy
-
-      form_hash_686c = get_form_hash_686c(file_number: bgs_person[:file_nbr].to_s)
-
-      BGS::SubmitForm686cJob.perform_async(uuid, claim.id, form_hash_686c) if claim.submittable_686?
-
+      bgs_person = service.people.find_person_by_ptcpnt_id(participant_id) || service.people.find_by_ssn(ssn) # rubocop:disable Rails/DynamicFindBy
+      file_number = bgs_person[:file_nbr]
+      # BGS's file number is supposed to be an eight or nine-digit string, and
+      # our code is built upon the assumption that this is the case. However,
+      # we've seen cases where BGS returns a file number with dashes
+      # (e.g. XXX-XX-XXXX). In this case specifically, we can simply strip out
+      # the dashes and proceed with form submission.
+      file_number = file_number.gsub('-', '') if file_number =~ /\A\d{3}-\d{2}-\d{4}\z/
```

Comment:
Is there a reason you don't want to just do
```ruby
file_number = bgs_person[:file_nbr].gsub('-', '')
```
instead of 

```ruby
      file_number = file_number.gsub('-', '') if file_number =~ /\A\d{3}-\d{2}-\d{4}\z/
```

Like, just always strip them if there are any and don't check for a regex?

Reply:
And also I think using `.tr` or `.delete` is better than gsub for a delete like this! Doesnt really matter though

```ruby
require 'benchmark'
n = 1_000_000
string = "055-444-285" 

Benchmark.bm do |x|
  x.report('gsub')           {  n.times {string.gsub(/-/, '')} }
  x.report('tr')             {  n.times {string.tr('-','')} }
  x.report('delete')         {  n.times {string.delete('-')} }
 end
````

```
       user     system      total        real
gsub  2.598974   0.022072   2.621046 (  2.661931)
tr  0.254542   0.001962   0.256504 (  0.257443)
delete  0.203386   0.001892   0.205278 (  0.206132)
```

Reply:
Yep! Reason there is that unless the file number is in the XXX-XX-XXXX format, I want to later log the exact format that we're getting (which may include dashes and god knows what else), so I can inform BGS and help them debug

Reply:
Just added in delete instead of gsub, thanks for that tip!